code/game/mecha/combat/combat.dm

```diff
@@ -3,10 +3,10 @@
 	internal_damage_threshold = 50
 	armor = list(melee = 30, bullet = 30, laser = 15, energy = 20, bomb = 20, bio = 0, rad = 0, fire = 100, acid = 100)
 
-/obj/mecha/combat/CheckParts(atom/holder)
+/obj/mecha/combat/CheckParts(list/parts_list)

```

Comment:
actually, this doesn't affect the parent at all.
however it is important to have the argument be up to date (it's always ```parts_list```, never an atom holder).